rest/server_context.go

```diff
@@ -515,16 +511,60 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(ctx context.Context, config
 					return nil, fmt.Errorf("attempting to create/update database with a scope/collection that is not found")
 				}
 
-				if err := initDataStore(dataStore); err != nil {
+				deferredIndexes, err := initDataStore(dataStore)
+				if err != nil {
 					return nil, err
 				}
+				deferredIndexesForDataStore[dataStore] = deferredIndexes
 			}
 		}
 	} else {
 		// no scopes configured - init the default data store
-		if err := initDataStore(bucket.DefaultDataStore()); err != nil {
+		deferredIndexes, err := initDataStore(bucket.DefaultDataStore())
+		if err != nil {
 			return nil, err
 		}
+		deferredIndexesForDataStore[bucket.DefaultDataStore()] = deferredIndexes
+	}
+
+	// Build and wait for Indexes to come online
+	wg := sync.WaitGroup{}
+	wg.Add(len(deferredIndexesForDataStore))
+	var buildError error
+
+	for dataStore, deferredIndexes := range deferredIndexesForDataStore {
+		func(dataStore sgbucket.DataStore, deferrdeferredIndexes []string) {
+			defer wg.Done()
+
+			if len(deferredIndexes) == 0 {
+				return
+			}
+
+			// ignoring second boolean return value since if datastore is not N1QLStore,
+			// it would have failed above while creating Indexes in `initDataStore` function
+			n1qlStore, _ := base.AsN1QLStore(dataStore)
+
+			err := base.BuildDeferredIndexes(n1qlStore, deferrdeferredIndexes)
```

Comment:
As discussed, should be able to do a single build for a bucket by changing the keyspace used in BuildDeferredIndexes to target the bucket.  Also need to tweak the keyspace used in the system:indexes query used to verify indexes are in deferred state before issuing the build - needs review whether can issue that at the bucket level as well.